### PR TITLE
Automate lambda list and revise lambda page

### PR DESCRIPTION
**Pull request by mooreds**
Fixes https://github.com/FusionAuth/fusionauth-site/issues/1121 and makes a few other improvements.

* generates the nav menu and overview list from the datafile
* added limits on how you can use liquid syntax in asciidoc to readme
* splitting _lambda-request and _lambda-response using a -suffix file because you can't have an include file have liquid syntax, so generating the list of valid lambda types was failing.
* fixes typos
* Adds clarifications for https://github.com/FusionAuth/fusionauth-issues/issues/613

### CODE REVIEW

File: site/docs/v1/tech/lambdas/index.adoc [hunk 1]
```diff
@@ -86,10 +87,33 @@ https://docs.oracle.com/javase/8/docs/technotes/guides/scripting/nashorn/
...
+----
+
+
+== Future Engines
 
-== Future engines
+The Nashorn engine is being phased out of Java in favor of more robust and advanced engines. 
 
-The Nashorn engine is being phased out of Java in favor of more robust and advanced engines. At any point in the future, we might switch from Nashorn to another JavaScript engine like V8. Therefore, use Nashorn features at your own risk.
+At any point in the future, we might switch from Nashorn to another JavaScript engine such as V8. Therefore, use Nashorn features at your own risk. Please https://github.com/FusionAuth/fusionauth-issues/issues/571[review this GitHub issue for more details].
```
Comment (robotdan) [inline site/docs/v1/tech/lambdas/index.adoc:119]: What does this mean exactly, can the user even use Nashorn features? Or do we just not know, so we are stating as a CYA?

Comment (mooreds) [reply site/docs/v1/tech/lambdas/index.adoc:119]: Brian wrote that 3 years ago: https://github.com/FusionAuth/fusionauth-site/blame/63aae4c24cfe91664a82af6d4de986baf1ecd777/site/docs/v1/tech/lambdas/index.adoc

Not sure what that means, other than maybe there are JS engine implementation details that are Nashorn specific?

File: DocsDevREADME.md [hunk 2]
```diff
@@ -280,6 +280,6 @@ Some sections are better suited to being driven by data. Jekyll makes this easy
 
 Examples of that are the customers page and the quotes widget.
 
-You can also go from asciidoc to liquid syntax. Examples of that are the 'related posts' section mentioned above, the themes form/api template docs, and the example apps.
+You can also go from asciidoc to liquid syntax. Examples of that are the 'related posts' section mentioned above, the themes form/api template docs, and the example apps. Note that you can't use liquid syntax in any include files. See https://github.com/asciidoctor/jekyll-asciidoc/issues/166 for more details about this issue.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply DocsDevREADME.md:283]: Me to! Was the cause of this tweet: https://twitter.com/mooreds/status/1468352584817647616

**Review by bguyza — APPROVED** [review]

**Comment by bguyza** [comment]
Looks good to me with exception of a reference that needs to be fixed.

File: site/docs/v1/tech/apis/lambdas.adoc [hunk 3]
```diff
@@ -142,14 +146,19 @@ include::docs/v1/tech/apis/_patch-support.adoc[]
 The unique Id of the Lambda to update.
 
 include::docs/v1/tech/apis/_lambda-put-request-body.adoc[]
+{% include docs/_lambda_type_api.liquid %}
+include::docs/v1/tech/apis/_lambda-put-request-body-suffix.adoc[]
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/apis/lambdas.adoc]: Awesome, thanks! I took a look and it turns out you can't change a lambda's type on `PUT` so we don't need the -suffix file or the `{%include` that I added.

**Pull request merged**